lib/renaissance/auctions/auctions.ex

```diff
@@ -0,0 +1,66 @@
+defmodule Renaissance.Auctions do
+  import Ecto.Query
+  alias Renaissance.{Auction, Repo, User, Users}
+
+  def create_auction(email, params) do
+    auction =
+      params["auction"]
+      |> add_user_id(email)
+      |> format_end_date()
+      |> format_price()
+
+    Repo.insert(Auction.changeset(%Auction{}, auction))
+  end
+
+  def get_by_title(title) do
+    Repo.get_by(Auction, title: title)
+  end
+
+  def get_all_auctions() do
```

Comment:
Rather than returning a list of an unspecified map here, I'd expect this to return `Auction` struct, with the seller preloaded:

```
Auction
|> preload(:user)
|> Repo.all()
```

This also leads to the mismatch between what you seem to want to call the `:user` association (`:seller`) and what you've called it on the model. In your ecto schema you could change the `:user` association to `:seller`:

`belongs_to :seller, User`

Then in User:

`has_many :auctions, Auction, foreign_key: :seller_id`